sdk/loadtestservice/load-testing-rest/src/getTestRunCompletionPoller.ts

```diff
@@ -0,0 +1,148 @@
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function getTestRunCompletionPoller(
+  client: AzureLoadTestingClient,
+  createTestRunResponse: TestRunCreateOrUpdate200Response | TestRunCreateOrUpdate201Response,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<TestRunCompletionPoller> {
+  type Handler = (state: OperationState<TestRunGet200Response>) => void;
+
+  const state: OperationState<TestRunGet200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestRunGet200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+  const testRunId = createTestRunResponse.body.testRunId;
+
+  const poller: SimplePollerLike<OperationState<TestRunGet200Response>, TestRunGet200Response> = {
+    async poll(options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      if (options?.abortSignal?.aborted) {
+        state.error = new Error("The operation was aborted.");
+        return;
+      }
+
+      if (testRunId) {
+        const getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+        if (isUnexpected(getTestRunResult)) {
+          state.status = "failed";
+          state.error = new Error(getTestRunResult.body.error.message);
+          return;
+        }
+
+        if (getTestRunResult.body.status === "FAILED") {
+          state.status = "failed";
+          state.error = new Error(getTestRunResult.body.status);
+        }
+
+        if (getTestRunResult.body.status === "CANCELLED") {
+          state.status = "canceled";
+        }
+
+        if (getTestRunResult.body.status === "DONE") {
+          state.status = "succeeded";
+        }
+
+        if (isTestRunInProgress(getTestRunResult.body)) {
+          state.status = "running";
+        }
+        state.result = getTestRunResult;
+        await processProgressCallbacks();
+      }
+    },
+
+    pollUntilDone(pollOptions?: { abortSignal?: AbortSignalLike }): Promise<TestRunGet200Response> {
+      return (resultPromise ??= (async () => {
+        const { abortSignal: inputAbortSignal } = pollOptions || {};
+        const { signal: abortSignal } = inputAbortSignal
+          ? new AbortController([inputAbortSignal, abortController.signal])
+          : abortController;
+        if (!poller.isDone()) {
+          await poller.poll({ abortSignal });
+          while (!poller.isDone()) {
+            const delay = sleep(currentPollIntervalInMs, abortSignal);
+            cancelJob = () => abortController.abort();
+            try {
+              await delay;
+            } catch (ex: any) {
+              state.status = "canceled";
+            }
```

Comment:
```suggestion
            await delay;
```

Reply:
The delay method throws an error from LRO which causes operation to stop, is there any other way we can do this.

Reply:
Updated with error thrown when timed out, as discussed offline